File: hrmsFe/src/app/model/getApplyAdvertisementResponse.ts
import {GetJobSeekerResponse} from "./getJobSeekerResponse";
import {GetAdvertisementResponse} from "./getAdvertisementResponse";

export interface GetApplyAdvertisementResponse {

  id: string,

  applyDate: Date,

  applyAdvertisementStatue: string,

  jobSeeker: GetJobSeekerResponse,

  advertisement: GetAdvertisementResponse
}

File: hrms/src/main/java/com/obss/hrms/response/GetApplyAdvertisementResponse.java
package com.obss.hrms.response;

import com.obss.hrms.entity.Advertisement;
import com.obss.hrms.entity.ApplyAdvertisementStatue;
import com.obss.hrms.entity.JobSeeker;

import java.time.LocalDate;

public record GetApplyAdvertisementResponse(
        String id,
        LocalDate applyDate,
        ApplyAdvertisementStatue applyAdvertisementStatue,
        JobSeeker jobSeeker,
        Advertisement advertisement
) {
}
